modules/internal/String.chpl

```diff
@@ -1419,18 +1427,18 @@ module String {
       }
       if boundsChecking {
         if r.hasLowBound() && (!r.hasHighBound() || r.size > 0) {
-          if r.low:int <= 0 then
-            halt("range out of bounds of string");
+          if r.low:int < 0 then
+            halt("range ", r, " out of bounds for string");
         }
       }
       // Loop to find whether the low and high codepoint indices
       // appear within the string.  Note the byte indices of those
       // locations, if they exist.
-      const cp_low = if r.hasLowBound() && r.low:int > 0 then r.low:int else 1;
-      const cp_high = if r.hasHighBound() then r.high:int else this.len + 1;
-      var cp_count = 1;
-      var byte_low = this.len + 1;  // empty range if bounds outside string
-      var byte_high = this.len;
+      const cp_low = if r.hasLowBound() && r.low:int >= 0 then r.low:int else 0;
+      const cp_high = if r.hasHighBound() then r.high:int else this.len;
```

Comment:
Note this line is also wrong and symmetric to our discussion yesterday. I would expect cp_high to be `this.len-1` here, for example.

Reply:
To be clear, it is not something that you should fix but something I should add to my branch here https://github.com/chapel-lang/chapel/compare/master...e-kayrakli:string-replace-better-check

Reply:
I think this was a bug with no symptoms. Other checks in the same code seem to prevent this to cause any issue. I'm fixing it regardless.